### DISCUSSION TITLE
How to reply before parseWithZod?

### DISCUSSION THREAD

**Discussion by bdrtsky** [edmundhung/conform#815] [Q&A]
I need to rate limit before I use Conform from server action. Of course I can return any object I want for my client, but then I can't make Typescript work with Conform. The only way I've found is to use `submission.reply` and put my rate limit error in `formErrors`. But I don't need to perform zod validation before this! Only after rate limit

```js
export async function requestOTPAction(_: unknown, formData: FormData) {
const formObject = Object.fromEntries(formData.entries())
if (process.env.NODE_ENV !== "development") {
    const headersList = await headers()
    const ip = headersList.get("x-forwarded-for")
  }
  const rateLimit = await limiter.check(rateLimitTries, ip!)
  if (rateLimit.status === "error") {
    // NEED TO RETURN RESPONSE HERE
  }
  
  const submission = parseWithZod(formData, {
    schema: EmailValidation,
  })
  if (submission.status !== "success") {
    return submission.reply({
      resetForm: false,
    })
  }
  return submission.reply()
}
```

Any way to use submission.reply() without actual zod validation? Any other way to handle custom responses?

**Comment by edmundhung** [comment]
It is not necessary to return a submission result back every time and just return null. Conform will ignore it and do nothing. If you want to pass along additional message, you can use the same pattern here: https://github.com/edmundhung/conform/discussions/816#discussioncomment-11252742

Having said that, I think this is a valid question if we care about progressive enhancement 🤔 definitely something I should consider for future api
